File: app/src/main/java/com/kirchhoff/example/githubclient/ui/splash/SplashPresenter.java
package com.kirchhoff.example.githubclient.ui.splash;

import android.support.annotation.NonNull;

import com.kirchhoff.example.githubclient.repository.keyvalue.KeyValueStorage;
import com.kirchhoff.example.githubclient.utils.TextUtils;

import java.util.concurrent.TimeUnit;

import rx.Observable;

/**
 * @author Kirchhoff-
 */

public class SplashPresenter implements SplashContract.Presenter {

    @NonNull
    private final KeyValueStorage storage;

    @NonNull
    private final SplashContract.View view;

    public SplashPresenter(@NonNull KeyValueStorage storage,
                           @NonNull SplashContract.View view) {
        this.storage = storage;
        this.view = view;
    }

    @Override
    public void subscribe() {
        Observable.just(true).delay(2, TimeUnit.SECONDS)
                .subscribe(aBoolean -> {
                    if (!TextUtils.isEmpty(storage.getUserName()))
                        view.openRepositoriesScreen();
                    else
                        view.openAuthScreen();
                });
    }
}

File: app/src/main/java/com/kirchhoff/example/githubclient/ui/splash/SplashPresenter.kt
package com.kirchhoff.example.githubclient.ui.splash

import com.kirchhoff.example.githubclient.repository.keyvalue.KeyValueStorage
import rx.Observable
import java.util.concurrent.TimeUnit

/**
 * @author Kirchhoff-
 */
class SplashPresenter : SplashContract.Presenter {

    private val storage: KeyValueStorage

    private val view: SplashContract.View

    constructor(storage: KeyValueStorage,
                view: SplashContract.View) {
        this.storage = storage
        this.view = view
    }

    override fun subscribe() {
        Observable.just(true).delay(2, TimeUnit.SECONDS)
                .subscribe {
                    if (storage.userName.isNullOrEmpty())
                        view.openAuthScreen()
                    else
                        view.openRepositoriesScreen()
                }
    }
}